### PR TITLE
[fix bug 1240862] Merge two user accounts into one.

### PR DESCRIPTION
**Pull request by akatsoulas**
* Migrate events to the current account.
* Migrate reports to the current account.
* Migrate event attendance to the current account.
* Migrate polls to the current account.
* Migrate user status to the current account.

~~* Delete old user.~~

### CODE REVIEW

**Comment by akatsoulas** [comment]
@johngian can you please r this?

**Comment by johngian** [comment]
Why do you prefer to do this with a database migrations rather than adding eg. a util function in /admin ? I don't really like the idea of using a migration for that. Also the code is going to be the same but more generic, meaning that we can also use this in the future.

**Comment by johngian** [comment]
Here is the list of FK and M2M referring to ``User`` not migrated:

* Poll
 * created_by
 * users_vote
* Vote
 * user
* PollComment
 * user
* EventComment
 * user
* UserProfile
 * added_by
 * mentor
* UserStatus
 * user
 * replacement_rep
* NGReport
 * mentor
* NGReportComment
 * user
* ActionItem
 * user
* FeaturedRep
 *  created_by
 *  user
* Bug
 * creator
 * assigned_to
 * cc
 * budget_needinfo

**Comment by akatsoulas** [comment]
I don't like the idea either. The reason for choosing to go this way is that merging accounts it's not something that it's supported and won't be in the future. The need for this migration comes from an error and it's one off. Specifically instead of re-activating and old account a new one was created.

Although this is going to be a destructive migration, in terms that the old account is going to be deleted, I removed the `delete()` method. If this migration run smoothly on prod, then we can manually delete the account. 

Regarding your comment about some missing `FK`s, I updated the migration to include the most important ones. A few of the are going to be deleted, due to the cascade delete, but I feel that it's the right way to go. Keeping eg action items from the old account only confusion is going to add. Moreover these kind of data won't affect the historic data.

Btw @johngian, pr updated :)

**Comment by akatsoulas** [comment]
cc @nukeador 

Data that we are going to migrate to the new profile:
* Events as creator
* Events attended
* Reports
* Votes
* Polls
* User statuses
* Mentees

Data that will be deleted:
* Action items
* Any kind of comment made
* Needinfo, assigned_to and cc from bugs.

**Comment by nukeador** [comment]
Seems good for me, great work guys! :)

File: remo/profiles/migrations/0060_merge_account.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+            current_profile = orm['profiles.UserProfile'].objects.get(
+                display_name=CURRENT_DISPLAY_NAME)
+        except orm['profiles.UserProfile'].DoesNotExist:
+            pass
+
+        if old_profile and current_profile:
+            # Re-assign reports to the current user
+            old_user = old_profile.user
+            current_user = current_profile.user
+            NGReport.objects.filter(user=old_user).update(user=current_user)
```
Comment (johngian) [inline remo/profiles/migrations/0060_merge_account.py]: Why are you using the model directly here and not the frozen ``orm[NGReport]``?

Comment (johngian) [inline remo/profiles/migrations/0060_merge_account.py]: These also has other occurences.

**Comment by johngian** [comment]
Code looks OK. r+wc
I also suggest to coordinate with a DBA to get a backup right before we release this on prod.

File: remo/profiles/migrations/0060_merge_account.py (hunk 1 above)
Comment (akatsoulas) [inline remo/profiles/migrations/0060_merge_account.py]: I tried this with frozen models and it does not work.

**Pull request merged**